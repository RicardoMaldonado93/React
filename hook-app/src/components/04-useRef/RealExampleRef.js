import React,{ useState } from 'react'
import { MultipleCustomHooks } from '../03-Examples/MultipleCustomHooks'

export const RealExampleRef = () => {

    const [show, setShow] = useState(false);




    return (
        <div>
            <h1>RealExampleRef</h1>
            <hr />

            { show && <MultipleCustomHooks />}

            <button 
                className="btn btn-outline-primary"
                onClick={()=>{setShow(!show)}}
            >
                { !show ? 'Mostrar' : 'Ocultar' }
            </button>
        </div>
    )
}
