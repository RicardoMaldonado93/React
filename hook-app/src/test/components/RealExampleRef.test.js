import React from 'react';
import {shallow} from 'enzyme';
import {RealExampleRef} from '../../components/04-useRef/RealExampleRef'

describe('Unitary test with RealExampleRef', () => {
    
    const wrapper = shallow( <RealExampleRef /> );

    test('should be correctly show', () => {
       expect( wrapper ).toMatchSnapshot();
       expect( wrapper.find("MultipleCustomHooks").exists()).toBeFalsy();
    });
    
   
    test('should be correctly show', () => {
        wrapper.find("button").simulate("click");
        expect( wrapper.find("MultipleCustomHooks").exists() ).toBeTruthy();
    });
   

});
