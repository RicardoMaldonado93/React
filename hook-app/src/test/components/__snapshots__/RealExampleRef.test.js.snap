// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unitary test with RealExampleRef should be correctly show 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    Mostrar
  </button>
</div>
`;
